src/activation.rs

```diff
@@ -161,10 +163,14 @@ pub fn receive_descriptors_with_names(
 fn socks_from_fds(num_fds: usize) -> Result<Vec<FileDescriptor>, SdError> {
     let mut descriptors = Vec::with_capacity(num_fds);
     for fd_offset in 0..num_fds {
-        let fd = SD_LISTEN_FDS_START + (fd_offset as i32);
-        let sock = FileDescriptor::try_from(fd)
-            .map_err(|e| format!("failed to receive file descriptor {}: {}", fd_offset, e))?;
-        descriptors.push(sock);
+        let index = SD_LISTEN_FDS_START
+            .checked_add(fd_offset as i32)
+            .ok_or_else(|| "overlarge file descriptor index")?;
+        let fd = FileDescriptor::try_from(index).unwrap_or_else(|(msg, val)| {
+            log::warn!("{}", msg);
```

Comment:
Contrarily to println-like functions, logging behavior is controlled by the consuming application, which is free to completely turn off logging from libraries.

See for some examples https://rust-lang-nursery.github.io/rust-cookbook/development_tools/debugging/config_log.html.

Reply:
Oh, that's good to know. Thanks :)

Meanwhile, since the `log` crate is coming in, I would really love to dump all those `LISTEN_foo` variables in the debug output.

Reply:
Ack, I can sprinkle a bunch of `log::trace!()` to trace-follow library behavior.
What are you interested in seeing? The raw (possibly missing) `env::var()` content or the values after parsing?

Reply:
I think the raw values would be nice to see.